qt/scientific_interfaces/Direct/ALFInstrumentModel.cpp

```diff
@@ -289,40 +190,109 @@ bool ALFInstrumentModel::addSelectedTube(DetectorTube const &tube) {
   return isNewTube;
 }
 
+bool ALFInstrumentModel::hasSelectedTubes() const { return !m_tubes.empty(); }
+
 bool ALFInstrumentModel::tubeExists(DetectorTube const &tube) const {
   return std::find(m_tubes.cbegin(), m_tubes.cend(), tube) != m_tubes.cend();
 }
 
-std::tuple<MatrixWorkspace_sptr, std::vector<double>>
-ALFInstrumentModel::generateOutOfPlaneAngleWorkspace(MantidQt::MantidWidgets::IInstrumentActor const &actor) const {
-  std::vector<double> twoThetas;
-  if (m_tubes.empty()) {
-    return {nullptr, twoThetas};
-  }
+std::unique_ptr<Mantid::API::AlgorithmRuntimeProps>
+ALFInstrumentModel::loadProperties(std::string const &filename) const {
+  auto properties = std::make_unique<AlgorithmRuntimeProps>();
+  AlgorithmProperties::update("Filename", filename, *properties);
+  AlgorithmProperties::update("OutputWorkspace", NOT_IN_ADS, *properties);
+  return properties;
+}
+
+std::unique_ptr<Mantid::API::AlgorithmRuntimeProps>
+ALFInstrumentModel::normaliseByCurrentProperties(Mantid::API::MatrixWorkspace_sptr const &inputWorkspace) const {
+  auto properties = std::make_unique<AlgorithmRuntimeProps>();
+  AlgorithmProperties::update("InputWorkspace", inputWorkspace, *properties);
+  AlgorithmProperties::update("OutputWorkspace", NOT_IN_ADS, *properties);
+  return properties;
+}
+
+std::unique_ptr<AlgorithmRuntimeProps> ALFInstrumentModel::rebinToWorkspaceProperties() const {
+  auto properties = std::make_unique<AlgorithmRuntimeProps>();
+  AlgorithmProperties::update("WorkspaceToRebin", m_vanadium, *properties);
+  AlgorithmProperties::update("WorkspaceToMatch", m_sample, *properties);
+  AlgorithmProperties::update("OutputWorkspace", NOT_IN_ADS, *properties);
+  return properties;
+}
```

Comment:
In general, I believe these `update` function calls are simply doing 
```
properties->setProperty(property_name, value);
```

They also do an extra check of the value in `value` to make sure its not an empty string or a nullptr (depending on what the type of `value` is). If its a string, I believe it might use the `setPropertyValue` function instead of `setProperty` 

In short I think the `AlgorithmProperties::update` provide a convenient way to safely set a parameter value within an IPropertyManager without having to worry about invalid values or which function to use.